lnwallet/channel_test.go

```diff
@@ -348,6 +339,26 @@ func TestSimpleAddSettleWorkflow(t *testing.T) {
 	}
 }
 
+// TestSimpleAddSettleWorkflow tests a simple channel scenario wherein the
+// local node (Alice in this case) creates a new outgoing HTLC to bob, commits
+// this change, then bob immediately commits a settlement of the HTLC after the
+// initial add is fully committed in both commit chains.
+//
+// TODO(roasbeef): write higher level framework to exercise various states of
+// the state machine
+//  * DSL language perhaps?
+//  * constructed via input/output files
+func TestSimpleAddSettleWorkflow(t *testing.T) {
+	t.Parallel()
+
+	for _, tweakless := range []bool{true, false} {
+		tweakless := tweakless
```

Comment:
It's actually a valid warning though IMO, albeit only when all the internal tests are _also_ being run in parallel.

Reply:
Did a quick test, because I wasn't sure about `t.Parallel` in combination with `t.Run`. Indeed, also sub tests can be ran in parallel, but only if `t.Parallel` is called in the sub test. Calling it in the root test doesn't make the sub tests run concurrently. In this case, afaik the sub test doesn't call `t.Parallel`, but it is a good precaution.